src/adagrams.js

```diff
@@ -1,15 +1,139 @@
-export const drawLetters = () => {
-  // Implement this method for wave 1
-};
+class Adagrams {
+  letterPool = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1,
+  };
 
-export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
-};
+  scoreChart = {
+    A: 1,
+    B: 3,
+    C: 3,
+    D: 2,
+    E: 1,
+    F: 4,
+    G: 2,
+    H: 4,
+    I: 1,
+    J: 8,
+    K: 5,
+    L: 1,
+    M: 3,
+    N: 1,
+    O: 1,
+    P: 3,
+    Q: 10,
+    R: 1,
+    S: 1,
+    T: 1,
+    U: 1,
+    V: 4,
+    W: 4,
+    X: 8,
+    Y: 4,
+    Z: 10,
+  };
 
-export const scoreWord = (word) => {
-  // Implement this method for wave 3
-};
+  drawLetters = () => {
+    // Implement this method for wave 1
+    let poolOfLetters = [];
 
-export const highestScoreFrom = (words) => {
-  // Implement this method for wave 1
-};
+    for (let [letter, count] of Object.entries(this.letterPool)) {
+      let multipleLetter = Array(count).fill(letter);
+      for (let letter of multipleLetter) {
+        poolOfLetters.push(letter);
+      }
+    }
+
+    let tenRandomLetters = [];
+
+    for (let i = 1; i <= 10; i++) {
+      let letter =
+        poolOfLetters[Math.floor(Math.random() * poolOfLetters.length)];
+      tenRandomLetters.push(letter);
+      poolOfLetters.splice(poolOfLetters.indexOf(letter), 1);
```

Comment:
👍 nice! this works just fine! an alternative strategy is to turn the randomly selected index value to `undefined` or a blank string once you've picked it. Then simply check if the next randomly selected index value is blank.

This might be a slightly more efficient suggestion, but it's a small change